### DISCUSSION TITLE
Basal/Pyramidal/Prismatic fractions

### DISCUSSION THREAD

**Discussion by hchaieb1234** [mtex-toolbox/mtex#1456] [Ask Anything] [answered]
Hello,

Could you please tell me how to extract statistics about basal/pyramidal/prismatic systems from ebsd data and plot them in a histogram ? (number of grains with a specific orientation) Thank you.

**Comment by kilir** [comment]
Hi,
can you elaborate what you mean by "systems"?
Cheers,
Rüdiger

**Reply by hchaieb1234** [reply]
In fact I would like to get a bar plot from EBSD data where I have the number of grains corresponding to a specific family basal/pyramidal I/pyramidal II/prismatic (I imagine there is a threshold to be defined)

An example is shown here for slip activation, but what I would like to get is the raw orientation if it is mainly basal or something else: 
![image](https://user-images.githubusercontent.com/74141009/173792087-ddc53d3e-d1e1-47a2-8223-b38f548a7c76.png)

**Reply by kilir** [reply]
Hi,
the bar plot is just the way of visualizing the count of grains which have a certain property. So please explain "a specific family " or "system" as before.
Cheers,
Rüdiger

**Reply by hchaieb1234** [reply]
Hi what I want to do is to determine the grains having an orientation that is close to a basal orientation (where the c axis is perpendicular to the analysed surface) and then do the same for other particular orientations (pyramidal/prismatic). The idea is to compare the orientations of grains (see if the c axis is mainly perpendicular to the surface or within the surface, I am working on HCP materials)

**Comment by kilir** [comment, marked as answer]
I see.
If `h` is the plane normal you'd like to have parallel to a reference direction `r` and that relation is given by orientation `o`, the direction of the reference direction in crystal coordinates is `hs = inv(o) * r`. The angle `angle(h,hs)` between `h` and `hs` tells you how "close" the two are.

Alternatively, `sh = o .*h.symmetrise` gives you (symmetrically equivalent) specimen directions belonging to `h` and `angle_outer(r,sh)` gives you all the possible angles between `r` and `sh` from which you can use the minimum value for each grains.

The latter possibility is probably less elegant.

In any case, you can then decide for a threshold of that angle to decide whether a plane is "close" to that reference or not and compute the ratio of the number of grains fulfilling this condition and the total number of all grains.

See also here:
https://mtex-toolbox.github.io/DefinitionAsCoordinateTransform.html#5
https://mtex-toolbox.github.io/DefinitionAsCoordinateTransform.html#10
https://mtex-toolbox.github.io/GrainOrientationParameters.html

Hope that helps.

Cheers,
Rüdiger
